From 04cbc5903b632aadedf8d17456c7541ce077e232 Mon Sep 17 00:00:00 2001
From: Colin McCabe <cmccabe@apache.org>
Date: Tue, 8 Jul 2014 18:17:28 +0000
Subject: [PATCH 518/791] HADOOP-10781. Unportable getgrouplist usage breaks
 FreeBSD (Dmitry Sivachenko via Colin Patrick
 McCabe)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1608871 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 96bfab8dbcba175c4eb3c9ede0550222b2ed4340)
---
 .../org/apache/hadoop/security/hadoop_user_info.c  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/security/hadoop_user_info.c b/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/security/hadoop_user_info.c
index 69c72a1..566da64 100644
--- a/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/security/hadoop_user_info.c
+++ b/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/security/hadoop_user_info.c
@@ -193,7 +193,7 @@ int hadoop_user_info_getgroups(struct hadoop_user_info *uinfo)
   ngroups = uinfo->gids_size;
   ret = getgrouplist(uinfo->pwd.pw_name, uinfo->pwd.pw_gid, 
                          uinfo->gids, &ngroups);
-  if (ret > 0) {
+  if (ret >= 0) {
     uinfo->num_gids = ngroups;
     ret = put_primary_gid_first(uinfo);
     if (ret) {
-- 
1.7.9.5

